packages/d3fc-chart/src/cartesian.js

```diff
@@ -58,8 +59,16 @@ export default (...args) => {
                 .attr('class', d => `x-label ${d}-label`)
                 .text(xLabel(data));
 
-            yLabelDataJoin(container, [yOrient(data)])
-                .attr('class', d => `y-label ${d}-label`)
+            const yOrientValue = yOrient(data);
+            const yLabelContainer = yContainerDataJoin(container, [yOrientValue])
+                .attr('class', d => `y-label-container ${d}-label`)
+                .style('grid-column', yOrientValue === 'left' ? 1 : 5)
+                .style('-ms-grid-column', yOrientValue === 'left' ? 1 : 5)
+                .style('grid-row', 3)
+                .style('-ms-grid-row', 3);
+
+            yLabelDataJoin(yLabelContainer, [yOrientValue])
```

Comment:
It feels like we shouldn't need to do both a yContainerDataJoin and then yLabelDataJoin?

For example, could you do:
```javascript
  // add the .y-label to new y-label containers
  yLabelContainer.enter().append('div').class('y-label');
  // Alter the text on the y-label
  yLabelContainer.select('.y-label').text(yLabel(data));
```

Reply:
I'll take a look, that didn't feel quite right.